hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -227,6 +227,7 @@ func (r *HostedClusterReconciler) managedResources() []client.Object {
 		&corev1.Namespace{},
 		&corev1.ServiceAccount{},
 		&corev1.Service{},
+		&corev1.Endpoints{},
```

Comment:
we should follow up to filter events for only "kubernetes" endpoint in enqueueHostedClustersFunc
then also consider only caching that resource by letting ObjectSelector to filte by field metadata.Name = "kubernetes"